_posts/2021-11-01-r-6.0.0.md

```diff
@@ -0,0 +1,139 @@
+
+```r
+library(arrow, warn.conflicts = FALSE)
+library(dplyr, warn.conflicts = FALSE)
+ 
+flights_filtered <- arrow_table(nycflights13::flights) %>%
+  select(carrier, origin, dest, arr_delay) %>%
+  # arriving early doesn't matter, so call negative delays 0
+  mutate(arr_delay = pmax(arr_delay, 0)) %>%
+  to_duckdb() %>%
+  # for each carrier-origin-dest, take the worst 5% of delays
+  group_by(carrier, origin, dest) %>%
+  mutate(arr_delay_rank = percent_rank(arr_delay)) %>%
+  filter(arr_delay_rank > 0.95)
+
+# pull data back into arrow to complete analysis
+flights_filtered %>%
+  to_arrow() %>%
+  # now summarise to get mean/min
+  group_by(carrier, origin, dest) %>%
+  summarise(arr_delay_mean = mean(arr_delay), arr_delay_min = min(arr_delay), num_flights = n()) %>%
+  filter(dest %in% c("ORD", "MDW")) %>%
+  arrange(desc(arr_delay_mean)) %>%
+  collect()
+```
+# Expanded use of altrep
+
+We are continuing our use of R’s altrep where possible. In 5.0.0 there were a limited set of circumstances that took advantage of altrep, but in 6.0.0 we have expanded types (to include strings), as well as vectors with `NA`s. 
+
+```r
+library(microbenchmark)
+library(arrow)
+
+tbl <-
+  arrow_table(data.frame(
+    x = rnorm(10000000),
+    y = sample(c(letters, NA), 10000000, replace = TRUE)
+  ))
+
+with_altrep <- function(data){
+  options(arrow.use_altrep = TRUE)
+  as.data.frame(data)  
+}
+
+without_altrep <- function(data){
+  options(arrow.use_altrep = FALSE)
+  as.data.frame(data)  
+}
+
+microbenchmark(
+  without_altrep(tbl),
+  with_altrep(tbl)
+)
+
+# Unit: milliseconds
+#                 expr      min        lq      mean    median        uq      max neval
+#  without_altrep(tbl) 191.0788 213.82235 249.65076 225.52120 244.26977 512.1652   100
+#     with_altrep(tbl)  48.7152  50.97269  65.56832  52.93795  55.24505 338.4602   100
+
+```
+# Over 30 new compute functions
+
+There are over 30 new compute functions available in this release, including string functions `str_to_lower()`, `str_to_upper()`, `str_to_title()`, `startsWith()`, `endsWith()`, `str_starts()`, and `str_ends()`, date and time functions `strftime()`, `format_ISO8601()`, `is_timestamp()` and others.
+
+# Can now install in an offline mode on linux
+
+For folks who need to install Arrow on an airgapped server, we have included helper functions and installation options that make it easier to download a fat-source of the arrow package that includes both the Arrow source as well as third-party dependencies that are needed when building Arrow.
+
+The helper function `create_package_with_all_dependencies()` can be run from a computer that does have access to the internet and it will create a fat-source package which can then be transferred and installed on a server without connectivity. This helper is also available on GitHub without installing the arrow package.  For more installation [see the docs](https://arrow.apache.org/docs/r/articles/install.html#offline-installation).
```

Comment:
```suggestion
The helper function `create_package_with_all_dependencies()` can be run from a computer that does have access to the internet and it will create a fat-source package which can then be transferred and installed on a server without connectivity. This helper is also available on GitHub without installing the arrow package.  For more installation [see the docs](https://arrow.apache.org/docs/r/articles/install.html#offline-installation).

Special thanks to Karl Dunkle Werner for the PRs to make this possible. Karl has also been added as a contributor in recognition of this contribution along with a number before this. Thank you, Karl!
```